dcc-storage-server/src/main/java/org/icgc/dcc/storage/server/repository/s3/S3UploadService.java

```diff
@@ -281,7 +273,14 @@ public void finalizeUpload(String objectId, String uploadId) {
         val objectKey = ObjectKeys.getObjectKey(dataDir, objectId);
         val request = new CompleteMultipartUploadRequest(actualBucketName, objectKey.getKey(), uploadId, etags);
 
-        s3Client.completeMultipartUpload(request);
+        try {
+          s3Client.completeMultipartUpload(request);
+        } catch (AmazonS3Exception e) {
+          if (e.getStatusCode() == HttpStatus.NOT_FOUND.value()) {
```

Comment:
What happens if this is not true? If there is a different error? This will suppress the exception. You need to bubble the exception up if you get an unexpected error. 

You should probably do a check of `!= NOT_FOUND` then `throw` otherwise log it with a `warn` and then continue with your short return.

Reply:
Sorry; that was a silly oversight. I took care of this in other places; but missed it here somehow. Fixing it now.